Race condition listing scm files.

From a [recent luci-go Travis build](https://travis-ci.org/luci/luci-go/builds/120364592):

```
panic: runtime error: index out of range
goroutine 1 [running]:
panic(0x66fb20, 0xc820012120)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/runtime/panic.go:464 +0x3e6
sort.(*StringSlice).Less(0xc8200eb0a0, 0x44, 0xffffffffffffffff, 0x1)
	<autogenerated>:13 +0x114
sort.doPivot(0x7fed0bd4f7e0, 0xc8200eb0a0, 0x44, 0x54, 0x3f, 0xa00)
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/sort/sort.go:128 +0x27b
... (Truncated for brevity) ...
	/home/travis/.gimme/versions/go1.6.linux.amd64/src/sort/sort.go:305 +0x86
github.com/maruel/pre-commit-go/scm.(*git).Between(0xc820017700, 0x6c8520, 0x9, 0x6c8530, 0x9, 0xc820142000, 0xa, 0xa, 0x0, 0x0, ...)
	/home/travis/gopath/src/github.com/maruel/pre-commit-go/scm/repo.go:343 +0xa36
main.(*application).cmdRunHook(0xc820013b10, 0x7fed0bd4b718, 0xc820017700, 0x6f3e80, 0x16, 0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/maruel/pre-commit-go/cmd/pcg/main.go:613 +0x23d
main.mainImpl(0x0, 0x0)
	/home/travis/gopath/src/github.com/maruel/pre-commit-go/cmd/pcg/main.go:822 +0x3b56
main.main()
	/home/travis/gopath/src/github.com/maruel/pre-commit-go/cmd/pcg/main.go:859 +0x24
```

The line in question is a simple `sort.Strings`, so it's difficult to believe that this can fail. However, there is [one case](https://github.com/maruel/pre-commit-go/blob/master/scm/repo.go#L294) where `files` equals `allFiles`, causing them to be sorted in parallel. As far as I can tell, this is the only possible source of a race here.

Recommend doing a simple "if files != allFiles" prior to launching the goroutine below.

--- a/scm/repo.go
+++ b/scm/repo.go
@@ -283,6 +283,7 @@ func (g *git) Between(recent, old Commit, ignorePatterns IgnorePatterns) (Change
 
 	// Gather list of changed files.
 	var files []string
+	filesEqualsAllFiles := false
 	if grecent == gitCurrent {
 		// Current is special cased, as it has to look at the checked out files.
 		go func() {
@@ -292,6 +293,7 @@ func (g *git) Between(recent, old Commit, ignorePatterns IgnorePatterns) (Change
 			// Fast path: diff against initial commit.
 			allFiles = <-allFilesCh
 			files = allFiles
+			filesEqualsAllFiles = true
 		} else {
 			// Gather list of unstaged file plus diff.
 			unstagedCh := make(chan []string)
@@ -335,11 +337,13 @@ func (g *git) Between(recent, old Commit, ignorePatterns IgnorePatterns) (Change
 
 	// Sort concurrently.
 	var wg sync.WaitGroup
-	wg.Add(1)
-	go func() {
-		defer wg.Done()
-		sort.Strings(files)
-	}()
+	if !filesEqualsAllFiles {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			sort.Strings(files)
+		}()
+	}
 	sort.Strings(allFiles)
 	wg.Wait()
 

